### PR TITLE
Snapshot Schedule Resource Implementation

### PR DESCRIPTION
**Pull request by nikitajoshi1**
# Description
  - Implement Snapshot Schedule Resource
  - Add Tests

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Feature Pull Request

##### RESOURCE OR DATASOURCE NAME
Snapshot Schedule Resource

##### OUTPUT

test

Coverage
Coverage2

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Unit tests
- [x] Acceptance tests

### CODE REVIEW

**Comment by nikitajoshi1** [comment]
run e2e test

File: examples/resources/powerscale_snapshot_schedule/resource.tf [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  # The path snapshotted. Default set to : "/ifs"
+  # path = "/ifs/tfacc_test"
+
+  # Time value in String for which snapshots created by this snapshot schedule should be retained. Values supported are of format : 
+  # "Never Expires, x Seconds(s), x Minute(s), x Hour(s), x Week(s), x Day(s), x Month(s), x Year(s) where x can be any integer value.
+  # Default set to : "1 Week(s)"
+  # retention_time = "3 Hour(s)" 
+
+  # Alias name to create for each snapshot.
+  # alias = "snap_alias"
```
Comment (doriac11) [inline examples/resources/powerscale_snapshot_schedule/resource.tf:38]: Does the Alias work correctly?

I noticed on snapshot even if you set the alias in the API it always just returns as `null` so I removed this field from snapshot.

Comment (nikitajoshi1) [reply examples/resources/powerscale_snapshot_schedule/resource.tf:38]: @doriac11 Yes it gives the value correctly if set.

**Comment by nikitajoshi1** [comment]
run e2e test

**Comment by nikitajoshi1** [comment]
run e2e test

**Comment by nikitajoshi1** [comment]
run e2e test

File: examples/resources/powerscale_snapshot_schedule/resource.tf (hunk 1 above)
Comment (doriac11) [reply examples/resources/powerscale_snapshot_schedule/resource.tf:38]: Okay thanks!

**Review by doriac11 — APPROVED** [review]

**Comment by nikitajoshi1** [comment]
run e2e test

**Comment by nikitajoshi1** [comment]
run e2e test

**Review by shekhar-j — APPROVED** [review]

**Pull request merged**